server/core/src/https/views/apps.rs

```diff
@@ -27,20 +25,53 @@ struct AppsView {
 #[derive(Template)]
 #[template(path = "apps_partial.html")]
 struct AppsPartialView {
-    // todo - actually list the applications the user can access here.
+    apps: Vec<AppLink>,
+}
+
+fn get_transformer(op_id: Uuid) -> impl Fn(OperationError) -> Response {
+    move |error| {
+        match error {
+            OperationError::SessionExpired => {
+                HtmlTemplate(LoginView {
+                    username: "",
+                    remember_me: false,
+                }).into_response()
+            }
+            _ => {
+                HtmlTemplate(UnrecoverableErrorView {
+                    err_code: error,
+                    operation_id: op_id,
+                }).into_response()
+            }
+        }
+    }
 }
 
 pub(crate) async fn view_apps_get(
-    State(_state): State<ServerState>,
-    Extension(_kopid): Extension<KOpId>,
+    State(state): State<ServerState>,
+    Extension(kopid): Extension<KOpId>,
     HxRequest(hx_request): HxRequest,
-    VerifiedClientInformation(_client_auth_info): VerifiedClientInformation,
+    VerifiedClientInformation(client_auth_info): VerifiedClientInformation,
 ) -> Response {
     // Because this is the route where the login page can land, we need to actually alter
     // our response as a result. If the user comes here directly we need to render the full
     // page, otherwise we need to render the partial.
+    let handler = get_transformer(kopid.eventid);
```

Comment:
I don't like this kind of weird factory/fn factory this returns,  Ithink there are simpler ways to achieve this.

Reply:
Yeah me neither, but I need to acquire the kopid for displaying errors somehow, switching to `axum::response::Result` does do away with the implementation here.

This is what I have right now but it's kinda the same issue
`.map_err(|old| { HtmxError::from((kopid, old)) })`